HLTrigger/Timer/interface/FastTimerService.h

```diff
@@ -86,690 +82,488 @@ class FastTimerService {
-  const double                                  m_dqm_moduletime_resolution;
-  std::string                                   m_dqm_path;
-
-  struct ProcessDescription {
-    std::string         name;
-    std::string         first_path;             // the framework does not provide a pre/postPaths or pre/postEndPaths signal,
-    std::string         last_path;              // so we emulate them keeping track of the first and last non-empty Path and EndPath
-    std::string         first_endpath;
-    std::string         last_endpath;
-    edm::ParameterSetID pset;
+  //void preModuleConstruction(edm::ModuleDescription const&);
+  //void postModuleConstruction(edm::ModuleDescription const&);
+
+  //void preModuleBeginJob(edm::ModuleDescription const&);
+  //void postModuleBeginJob(edm::ModuleDescription const&);
+
+  //void preModuleEndJob(edm::ModuleDescription const&);
+  //void postModuleEndJob(edm::ModuleDescription const&);
+
+  //void preModuleBeginStream(edm::StreamContext const&, edm::ModuleCallingContext const&);
+  //void postModuleBeginStream(edm::StreamContext const&, edm::ModuleCallingContext const&);
+
+  //void preModuleEndStream(edm::StreamContext const&, edm::ModuleCallingContext const&);
+  //void postModuleEndStream(edm::StreamContext const&, edm::ModuleCallingContext const&);
+
+  void preModuleGlobalBeginRun(edm::GlobalContext const&, edm::ModuleCallingContext const&);
+  void postModuleGlobalBeginRun(edm::GlobalContext const&, edm::ModuleCallingContext const&);
+
+  void preModuleGlobalEndRun(edm::GlobalContext const&, edm::ModuleCallingContext const&);
+  void postModuleGlobalEndRun(edm::GlobalContext const&, edm::ModuleCallingContext const&);
+
+  void preModuleGlobalBeginLumi(edm::GlobalContext const&, edm::ModuleCallingContext const&);
+  void postModuleGlobalBeginLumi(edm::GlobalContext const&, edm::ModuleCallingContext const&);
+
+  void preModuleGlobalEndLumi(edm::GlobalContext const&, edm::ModuleCallingContext const&);
+  void postModuleGlobalEndLumi(edm::GlobalContext const&, edm::ModuleCallingContext const&);
+
+  void preModuleStreamBeginRun(edm::StreamContext const&, edm::ModuleCallingContext const&);
+  void postModuleStreamBeginRun(edm::StreamContext const&, edm::ModuleCallingContext const&);
+
+  void preModuleStreamEndRun(edm::StreamContext const&, edm::ModuleCallingContext const&);
+  void postModuleStreamEndRun(edm::StreamContext const&, edm::ModuleCallingContext const&);
+
+  void preModuleStreamBeginLumi(edm::StreamContext const&, edm::ModuleCallingContext const&);
+  void postModuleStreamBeginLumi(edm::StreamContext const&, edm::ModuleCallingContext const&);
+
+  void preModuleStreamEndLumi(edm::StreamContext const&, edm::ModuleCallingContext const&);
+  void postModuleStreamEndLumi(edm::StreamContext const&, edm::ModuleCallingContext const&);
+
+  void preModuleEventPrefetching(edm::StreamContext const&, edm::ModuleCallingContext const&);
```

Comment:
Before running a module, the framework will prefetch all data the module says it will consume. The `preModuleEventPrefetching` happens when the framework has decided the module should be run and it is going to start prefetching its data. `postModuleEventPrefetching` is signaled when all the prefetching has completed and the framework can now schedule to run the module. For global and stream module, the framework will start them immediately. For legacy and one modules, the framework will wait until it can obtain the shared resource.